internal/alb/rs/rule.go

```diff
@@ -324,3 +318,43 @@ func (r Rule) IsDesiredDefault() bool {
 	}
 	return *r.rs.desired.IsDefault
 }
+
+func (r Rule) valid(listenerPort int64, listenerProtocol *string) bool {
+	if r.rs.desired.Actions[0].RedirectConfig != nil {
+		var host, path *string
+		rc := r.rs.desired.Actions[0].RedirectConfig
+
+		for _, c := range r.rs.desired.Conditions {
+			if *c.Field == "host-header" {
+				host = c.Values[0]
+			}
+			if *c.Field == "path-pattern" {
+				path = c.Values[0]
+			}
+		}
+
+		if host == nil && *rc.Host != "#{host}" {
```

Comment:
i think the two if for host check can be simplified to be *rc.Host != "#{host}" , since
host can never be "#{host}", according to k8s spec, it need to be a domain name of RFC 3986.